### PR TITLE
Added is present check to FreeSymbolsVisitor

### PR DESCRIPTION
**Pull request by npochhi**
Relevant: #1386

### CODE REVIEW

File: symengine/visitor.cpp [hunk 1]
```diff
@@ -73,10 +73,14 @@ class FreeSymbolsVisitor : public BaseVisitor<FreeSymbolsVisitor>
...
     set_basic s;
+    uset_basic v;
 
     void bvisit(const Symbol &x)
     {
-        s.insert(x.rcp_from_this());
+        if (v.find(x.rcp_from_this()) == v.end()) {
+            v.insert(x.rcp_from_this());
+            s.insert(x.rcp_from_this());
+        }
```
Comment (isuruf) [inline symengine/visitor.cpp]: This is not really necessary and will be a bit slow.

File: symengine/visitor.cpp [hunk 2]
```diff
@@ -87,14 +91,20 @@ class FreeSymbolsVisitor : public BaseVisitor<FreeSymbolsVisitor>
         }
         s.insert(set_.begin(), set_.end());
         for (const auto &p : x.get_point()) {
-            p->accept(*this);
+            if (v.find(p->rcp_from_this()) == v.end()) {
+                v.insert(p->rcp_from_this());
```
Comment (isuruf) [inline symengine/visitor.cpp]: This is a bit slow. See https://stackoverflow.com/a/10163885/4768820

**Comment by npochhi** [comment]
ping @isuruf

**Comment by npochhi** [comment]
ping @isuruf

**Comment by isuruf** [comment]
I removed the last commit, which was unneeded and merged. Thanks @npochhi

**Pull request merged**